### ISSUE TITLE
Delete rows by date value.

### ISSUE THREAD

**Issue by bsurai** [typeorm/typeorm#1051]
I want to build delete-query like that. But it does not work for me. How can I do that? What approach is better?

```
import {getConnection} from "typeorm";
await getConnection()
    .createQueryBuilder()
    .delete()
    .from(User)
    .where("dateTime <= :date_param",  { date_param: new Date()  })
    .execute();
```

**Comment by pleerock** [comment]
Please format your code properly. What it does not work for you? What error do you have? Please provide more info.

**Comment by bsurai** [comment]
I don't have error. It just does not delete old records.
My table has format like that.

```
import {Entity, PrimaryGeneratedColumn, Column} from 'typeorm';

@Entity()
export class User  {

	@PrimaryGeneratedColumn()
	id: number;

	@Column()
	billingTag: string;

	@Column()
	airtimeInSeconds: number;

	@Column()
	dateTime: Date;
}
```

**Comment by bsurai** [comment]
When I tried use `dateTime` as a time stamp, it worked good. I mean:

```
@Column()
	dateTime: namber;
```
and

`.where("dateTime <= :date_param",  { date_param: Date.now()  })`

**Comment by bsurai** [comment]
But I'd like to use Date time. So I deleted old database and made new schema

**Comment by pleerock** [comment]
Can you enable query logging and show what query it executes?

**Comment by bsurai** [comment]
Here is logs from app starting, deleting old rows, creating new row, checking existing rows.

```
executing query:  BEGIN TRANSACTION
executing query:  SELECT * FROM sqlite_master WHERE type = 'table' AND name IN ('usage_report', 'usage_report')
executing query:  PRAGMA table_info("usage_report")
executing query:  PRAGMA index_list("usage_report")
executing query:  PRAGMA foreign_key_list("usage_report")
executing query:  CREATE TABLE "temporary_usage_report" ("id" integer NOT NULL PRIMARY KEY AUTOINCREMENT, "billingTag" varchar NOT NULL, "airtimeInSeconds" integer NOT NULL, "dateTime" datetime NOT NULL, "sent" boolean NOT NULL)
executing query:  INSERT INTO "temporary_usage_report"("id", "billingTag", "airtimeInSeconds", "dateTime", "sent") SELECT "id", "billingTag", "airtimeInSeconds", "dateTime", "sent" FROM "usage_report"
executing query:  DROP TABLE "usage_report"
executing query:  ALTER TABLE "temporary_usage_report" RENAME TO "usage_report"
executing query:  CREATE TABLE "temporary_usage_report" ("id" integer NOT NULL PRIMARY KEY AUTOINCREMENT, "billingTag" varchar NOT NULL, "airtimeInSeconds" integer NOT NULL, "dateTime" integer NOT NULL, "sent" boolean NOT NULL)
executing query:  INSERT INTO "temporary_usage_report"("id", "billingTag", "airtimeInSeconds", "dateTime", "sent") SELECT "id", "billingTag", "airtimeInSeconds", "dateTime", "sent" FROM "usage_report"
executing query:  DROP TABLE "usage_report"
executing query:  ALTER TABLE "temporary_usage_report" RENAME TO "usage_report"
executing query:  COMMIT
start
2017-10-20T07:22:46.050Z
executing query:  **DELETE FROM** "usage_report" WHERE "dateTime" <= $1 -- PARAMETERS: ["2017-10-20T07:22:46.050Z"]
delete: 24.167ms
END
Inserting a new record into the database...
executing query:  BEGIN TRANSACTION
executing query:  INSERT INTO "usage_report"("billingTag", "airtimeInSeconds", "dateTime", "sent") VALUES ($1,$2,$3,$4) -- PARAMETERS: ["Timber",10000,"2017-10-20 07:22:46.076",0]
executing query:  COMMIT
Saved a new record with id: 17
Loading report from the database...
executing query:  SELECT "UsageReport"."id" AS "UsageReport_id", "UsageReport"."billingTag" AS "UsageReport_billingTag", "UsageReport"."airtimeInSeconds" AS "UsageReport_airtimeInSeconds", "UsageReport"."dateTime" AS "UsageReport_dateTime", "UsageReport"."sent" AS "UsageReport_sent" FROM "usage_report" "UsageReport"
Loaded report:  [ UsageReport {
    billingTag: 'Timber',
    airtimeInSeconds: 10000,
    dateTime: 2017-10-20T06:58:31.116Z,
    sent: false,
    id: 1 },
  UsageReport {
    billingTag: 'Timber',
    airtimeInSeconds: 10000,
    dateTime: 2017-10-20T07:00:27.231Z,
    sent: false,
    id: 2 },
  UsageReport {
    billingTag: 'Timber',
    airtimeInSeconds: 10000,
    dateTime: 2017-10-20T07:00:44.269Z,
    sent: false,
    id: 3 },
....

```

**Comment by bsurai** [comment]
@pleerock hi. 
I've found a solution. I can use sqlite date/time function. In my case I use 
```
const OneWeek = 7;
.....
.where(`dateTime < date('now', 'start of day', '-${OneWeek} day')`)
....
```

**Comment by pleerock** [comment]
This issue is fixed in 0.1.3
